app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,373 @@
+            let width = style.get_number(theme::separator::width);
+            let height = style.get_number(theme::separator::height);
+            let triangle = Triangle(width, height).rotate((PI/2.0).radians());
+            let shape = triangle.fill(color);
+            shape.into()
+        }
+    }
+}
+
+/// A three dots icon. Used to indicate that the last module in breadcrumbs list contains more
+/// ancestors.
+pub mod ellipsis {
+    use super::*;
+
+    pub const ICON_WIDTH: f32 = 32.0;
+
+    ensogl_core::define_shape_system! {
+        above = [ensogl_grid_view::entry::shape];
+        pointer_events = false;
+        (style: Style, alpha: f32) {
+            let radius = style.get_number(theme::ellipsis::circles_radius).px();
+            let gap = style.get_number(theme::ellipsis::circles_gap).px();
+            let background_width = style.get_number(theme::ellipsis::background_width);
+            let background_height = style.get_number(theme::ellipsis::background_height);
+            let background_corners_radius = style.get_number(theme::ellipsis::background_corners_radius);
+            let col = style.get_color(theme::ellipsis::circles_color);
+            let circles_color = Var::<color::Rgba>::rgba(col.red,col.green,col.blue,alpha.clone());
+            let bg_col = style.get_color(theme::ellipsis::background_color);
+            let background_color = Var::<color::Rgba>::rgba(bg_col.red,bg_col.green,bg_col.blue,alpha);
+
+            // TODO: `.repeat()` call would be more efficient than three separate circles, but it
+            //   also requires `.intersection()` to crop the infinite grid of circles, which is
+            //   currently broken. See https://www.pivotaltracker.com/story/show/182593513.
+            let left = Circle(radius.clone()).fill(circles_color.clone());
+            let center = Circle(radius.clone()).fill(circles_color.clone());
+            let right = Circle(radius).fill(circles_color);
+            let circles = left.translate_x(-gap.clone()) + center + right.translate_x(gap);
+            let background = Rect((background_width.px(), background_height.px()));
+            let background = background.corners_radius(background_corners_radius.px());
+            let background = background.fill(background_color);
+            let shape = background + circles;
+            shape.into()
+        }
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+/// A model for the entry in the breadcrumbs list.
+#[allow(missing_docs)]
+#[derive(Clone, CloneRef, Debug, Default)]
+pub enum Model {
+    #[default]
+    Ellipsis,
+    Text(ImString),
+    Separator,
+}
+
+#[allow(missing_docs)]
+#[derive(Clone, Copy, CloneRef, Debug, Default, PartialEq)]
+enum State {
+    #[default]
+    Ellipsis,
+    Text,
+    Separator,
+}
+
+
+
+// =============
+// === Entry ===
+// =============
+
+// === EntryData ===
+
+/// An internal structure of [`Entry`], which may be passed to FRP network.
```

Comment:
1. Does this doc tells that this structure may be passed to FRP, or that Entry can be passed to FRP? Please make sentences in docs more obvious (here in other places - please check where doc sentences can be shorter and make sure that the subject is not ambiguous).
2. If this tells that this object can be passed to FRP, the clone here is pretty costly - it has 6 or 7 Rcs to be cloned. In such a case, this sohudl be wrapped in one Rc.

Reply:
1. Rephrased the documentation completely.
2. It is wrapped in a single Rc (see `Entry` struct implementation)